src/org/testKitGen/UtilsGen.java

```diff
@@ -15,20 +15,22 @@
 package org.testKitGen;
 
 import java.io.File;
-import java.io.FileWriter;
 import java.io.IOException;
+import java.io.Writer;
 import java.util.List;
 
 public class UtilsGen {
 	private Arguments arg;
 	private ModesDictionary md;
+	private MkGen mg;
 	private String utilsMk;
 	private String rerunMk;
 	private List<String> ignoreOnRerunList;
 
 	public UtilsGen(Arguments arg, ModesDictionary md, List<String> ignoreOnRerunList) {
 		this.arg = arg;
 		this.md = md;
+		this.mg = new MkGen(arg, null, null, rerunMk, ignoreOnRerunList, ignoreOnRerunList);
```

Comment:
- It doesn't seem right to initialize `MkGen` solely to use `getWriterObject()` in `MkGen.java`, especially when it involves setting fake parameters in `MkGen`.

- `getWriterObject()` should be a utility function that is leveraged by all files, including EnvDetector.java.

Reply:
Please remove `this.mg = new MkGen(arg, null, null, rerunMk, ignoreOnRerunList, ignoreOnRerunList);`. It is not used.